services/antivirus/pkg/service/service.go

```diff
@@ -16,34 +16,44 @@ import (
 	"github.com/cs3org/reva/v2/pkg/events"
 	"github.com/cs3org/reva/v2/pkg/events/stream"
 	"github.com/cs3org/reva/v2/pkg/rhttp"
+	"go.opentelemetry.io/otel/trace"
+
 	"github.com/owncloud/ocis/v2/ocis-pkg/log"
 	"github.com/owncloud/ocis/v2/services/antivirus/pkg/config"
 	"github.com/owncloud/ocis/v2/services/antivirus/pkg/scanners"
-	"go.opentelemetry.io/otel/trace"
 )
 
 var (
-	// ErrFatal is returned when a fatal error occurs and we want to exit.
+	// ErrFatal is returned when a fatal error occurs, and we want to exit.
 	ErrFatal = errors.New("fatal error")
 	// ErrEvent is returned when something went wrong with a specific event.
 	ErrEvent = errors.New("event error")
 )
 
 // Scanner is an abstraction for the actual virus scan
 type Scanner interface {
-	Scan(file io.Reader) (scanners.ScanResult, error)
+	Scan(body scanners.Input) (scanners.Result, error)
 }
 
 // NewAntivirus returns a service implementation for Service.
 func NewAntivirus(c *config.Config, l log.Logger, tp trace.TracerProvider) (Antivirus, error) {
-	av := Antivirus{c: c, l: l, tp: tp, client: rhttp.GetHTTPClient(rhttp.Insecure(true))}
 
+	var scanner Scanner
 	var err error
-	av.s, err = scanners.New(c)
+	switch c.Scanner.Type {
+	default:
+		return Antivirus{}, fmt.Errorf("unknown av scanner: '%s'", c.Scanner.Type)
+	case "clamav":
+		scanner = scanners.NewClamAV(c.Scanner.ClamAV.Socket)
+	case "icap":
+		scanner, err = scanners.NewICAP(c.Scanner.ICAP.URL, c.Scanner.ICAP.Service, time.Duration(c.Scanner.ICAP.Timeout)*time.Second)
```

Comment:
We should change config value to be of type `time.Duration`. (Needs deprecation maybe.) That way users could define the timeout like `30s`. What do you think?

Reply:
yup, needs further polishing... happy to take care once we see it does what it should